SCClassLibrary/Common/Collections/SequenceableCollection.sc

```diff
@@ -111,8 +111,10 @@ SequenceableCollection : Collection {
 	asSequenceableCollection { ^this }
 
 	// select an element at random
-	choose {
-		^this.at(this.size.rand)
+	choose { arg weights=nil;
+		if (weights.isNil)
+			{^this.at(this.size.rand)}
+			{this.wchoose(weights.normalizeSum)};
```

Comment:
You are proposing to change a method that many people use all the time. You should at least check if it works! This code doesn't.

Also it would be good if you keep in line with the standard formatting (look around in the same file).

Reply:
Certainly!